### DISCUSSION TITLE
Package manager broke because repo zipballs have recently changed SHA

### DISCUSSION THREAD

**Discussion by Janiczek** [community/community#24652] [Apps, API and Webhooks]
<p>Hello!</p>
<p>We have started having a problem in the Elm language community today where the freshly downloaded repo zipball SHA1 hash no longer agrees with SHA1 in the Elm package repository, which bricks the package manager.</p>
<p>Example:<br>
<a href="https://github.com/chain-partners/elm-bignum/zipball/1.0.1/" rel="noopener nofollow ugc">https://github.com/chain-partners/elm-bignum/zipball/1.0.1/</a> is expected to have a SHA1 hash <code>77f80b44ceef54982b0f161753d029412ae442e0</code> but downloading the zipball and running <code>shasum</code> on it results in <code>37c353f68e484a16c482bd545b680cd5dd9d9eaa</code> :x:</p>
<p>Note that this doesn’t apply across all repositories:<br>
<a href="https://github.com/elm/core/zipball/1.0.5/" rel="noopener nofollow ugc">https://github.com/elm/core/zipball/1.0.5/</a> is expected to have a SHA1 hash <code>9288a7574b778b4ebc6557d504a0b16c09daab43</code> and that’s what <code>shasum</code> gives too :white_check_mark:</p>
<p>Any idea what caused this change? Has something been deployed in GitHub recently that changed this behaviour?</p>
<p>Should we just adjust to the new hashes or would this be rolled back? Can we rely on the zipballs not changing content (if the repo didn’t change)?</p>

**Comment by Janiczek** [comment]
<p>Here’s a relevant comment from somewhere else: <a href="https://github.com/keybase/client/issues/10800#issuecomment-377417315" class="inline-onebox" rel="noopener nofollow ugc">Github generated tarballs change randomly · Issue #10800 · keybase/client · GitHub</a></p>

**Comment by asv** [comment]
<p>ZIPballs have different modtimes for files by 5 hours.</p>
<p>There are three hard things in Computer Science: cache invalidation, naming things and date&amp;time (UTC) things.</p>

**Comment by Janiczek** [comment, marked as answer]
<aside class="onebox twitterstatus" data-onebox-src="https://twitter.com/tgummerer/status/1488493440103030787">
  <header class="source">

      <a href="https://twitter.com/tgummerer/status/1488493440103030787" target="_blank" rel="noopener nofollow ugc">twitter.com</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/149425/181085846-00702ea9-799a-4623-a5b6-637df5a75a20.jpeg" class="thumbnail onebox-avatar" width="400" height="400">
<h4><a href="https://twitter.com/tgummerer/status/1488493440103030787" target="_blank" rel="noopener nofollow ugc">Thomas Gummerer</a></h4>
<div class="twitter-screen-name"><a href="https://twitter.com/tgummerer/status/1488493440103030787" target="_blank" rel="noopener nofollow ugc">@tgummerer</a></div>

<div class="tweet">
  <span class="tweet-description">@shs96c @github @bazelbuild Engineer on the team responsible at GitHub here.  I just noticed this, and we rolled back the feature flag that we believe was causing the change, and will look into making the change without breaking the checksums.</span>
</div>

<div class="date">
  <a href="https://twitter.com/tgummerer/status/1488493440103030787" class="timestamp" target="_blank" rel="noopener nofollow ugc">3:51 AM - 1 Feb 2022</a>

    <span class="like">
      <svg viewbox="0 0 512 512" width="14px" height="16px" aria-hidden="true">
        <path d="M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z"></path>
      </svg>
      6
    </span>

</div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by JamesAtIntegratnIO** [comment]
<p>Is this endpoint being worked on again? I’m getting sha256 mismatches today.<br>
what I expect:</p>
<pre><code class="lang-auto">      targetDir = "";
      src = composerEnv.buildZipPackage {
        name = "therobfonz-laravel-mandrill-driver-d4f59dea7832abd2b023cac4245eef21d6174a7c";
        src = fetchurl {
          url = "https://api.github.com/repos/therobfonz/laravel-mandrill-driver/zipball/d4f59dea7832abd2b023cac4245eef21d6174a7c";
          sha256 = "1jjbgq10p9md14ks8r3q0k62vrnc45fr3nvq2xmkm2p8c2fz31r5";
        };
      };
    };
</code></pre>
<p>What I’m getting:</p>
<pre><code class="lang-auto">path is '/nix/store/2wd6ga5a1pk985262bm31yqi87v0wdc9-d4f59dea7832abd2b023cac4245eef21d6174a7c'
0ik395i6gdfs640pjx0mm4zy8x44mwqvbchzcfch0579b9m12fn6
</code></pre>
